cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/JmesPathOutputFilter.cs

```diff
@@ -3,25 +3,38 @@
 
 namespace Microsoft.Kiota.Cli.Commons.IO;
 
+/// <summary>
+/// An output filter that uses JMESPath queries to filter the output
+/// </summary>
 public class JmesPathOutputFilter : IOutputFilter
 {
     private readonly JmesPath jmesPath;
 
+    /// <summary>
+    /// Creates a new instance of JmesPathOutputFilter
+    /// </summary>
+    /// <param name="jmesPath">The JmesPath transformer instance.</param>
     public JmesPathOutputFilter(JmesPath jmesPath)
     {
         if (jmesPath is null) throw new ArgumentNullException(nameof(jmesPath), $"Parameter '{nameof(jmesPath)}' is required.");
         this.jmesPath = jmesPath;
     }
 
-    public Stream FilterOutput(Stream content, string query) {
+    /// <inheritdoc />
+    public async Task<Stream> FilterOutputAsync(Stream content, string query, CancellationToken cancellationToken = default) {
         if (string.IsNullOrEmpty(query)) return content;
+        cancellationToken.ThrowIfCancellationRequested();
         using var reader = new StreamReader(content);
-        var strContent = reader.ReadToEnd();
+        var strContent = await reader.ReadToEndAsync();
```

Comment:
```suggestion
        var strContent = await reader.ReadToEndAsync(cancellationToken);
```

Reply:
I don't think that function is cancellable. There's no function overload that accepts a cancellation token

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader.readtoendasync?view=net-6.0